### PR TITLE
POD-7459 Update and Build Overcoat

### PR DESCRIPTION
**Pull request by cjheng**
This PR updates the dependencies of this repo and updates `MTLManagedObjectAdapter` so that it's buildable and testable in Xcode 12.

- The `carthage-update.sh` script has been introduced to allow Carthage dependencies to be build with Xcode 12.
- The `void` keyword was added in `MTLManagedObjectAdapter/extobjc/EXTScope.h` to address an instance of the "This block declaration is not a prototype" error.
- `NSConfinementConcurrencyType` has been replaced with `NSPrivateQueueConcurrencyType` as the former is deprecated and unavailable.
- The `attributes` variable has been freed in one of the early returns in `MTLManagedObjectAdapter/extobjc/EXTRuntimeExtensions.m` to address a potential memory leak warning.
- Xcode recommended project settings have been applied.

### CODE REVIEW

File: MTLManagedObjectAdapter.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -58,6 +58,12 @@
 /* End PBXContainerItemProxy section */
 
 /* Begin PBXFileReference section */
+		3F6F6B412538EB6D00F25078 /* macOS-Application.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = "macOS-Application.xcconfig"; sourceTree = "<group>"; };
```
Comment (nservidio) [inline MTLManagedObjectAdapter.xcodeproj/project.pbxproj:61]: Did all the changes in this file get introduced when you updated to recommended project settings?

**Review by nservidio — COMMENTED** [review]
Looks good! Just one question.

File: MTLManagedObjectAdapter.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -569,40 +592,44 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 5470B58C198514F700E4D941 /* Release.xcconfig */;
 			buildSettings = {
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = NO;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (nservidio) [inline MTLManagedObjectAdapter.xcodeproj/project.pbxproj:596]: This looks like it already drops support for 10 and 11. Is https://ovuline.atlassian.net/browse/POD-8610 in the queue?

File: MTLManagedObjectAdapter.xcodeproj/project.pbxproj (hunk 1 above)
Comment (cjheng) [reply MTLManagedObjectAdapter.xcodeproj/project.pbxproj:61]: Yeah, the only changes I manually made were listed in the PR description. Everything else like these pbxproj changes were via Xcode's recommended project settings.

Edit: I may have had to manually specify the Swift version myself... I don't recall if that was actually done for this PR or when I was experimenting with other libraries.

File: MTLManagedObjectAdapter.xcodeproj/project.pbxproj (hunk 2 above)
Comment (cjheng) [reply MTLManagedObjectAdapter.xcodeproj/project.pbxproj:596]: Looks like it's still in to be groomed. POD-8612 has also been moved to next sprint due to other priorities needing to be handled in the current sprint. It shouldn't be an issue to handle POD-8610 before it if we have to.

**Pull request merged**